apax/nn/models.py

```diff
@@ -79,8 +79,10 @@ def __call__(
             perturbation,
         )
 
-        gm = self.descriptor(dr_vec, Z, idx)
-        features = jax.vmap(self.readout)(gm)
+        features = self.descriptor(dr_vec, Z, idx)
+
+        if self.readout:
+            features = jax.vmap(self.readout)(features)
```

Comment:
I can't say anything about this as I don't know what exactly it is doing. I guess this is the fix.

Reply:
the readout layer is now optional, enabling just the evaluation of the descriptor.